From fcaecd88457f6153879c79f2e961fb33bf47cb4a Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 21 Sep 2022 12:19:27 +0300
Subject: [PATCH 02/30] libc: Correct PRIxPTR define

The size of uintptr_t is different on arm64 and arm32 platforms,
so PRIxPTR must be defined differently based on address size.

Issue: ALB-9298
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/lib/libc/aarch32/inttypes_.h | 6 ++++++
 include/lib/libc/aarch64/inttypes_.h | 7 +++++++
 include/lib/libc/inttypes.h          | 6 ------
 3 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/include/lib/libc/aarch32/inttypes_.h b/include/lib/libc/aarch32/inttypes_.h
index 11d2d3525..ca218f7fa 100644
--- a/include/lib/libc/aarch32/inttypes_.h
+++ b/include/lib/libc/aarch32/inttypes_.h
@@ -17,5 +17,11 @@
 #define PRIu64		"llu"	/* uint64_t */
 #define PRIx64		"llx"	/* uint64_t */
 #define PRIX64		"llX"	/* uint64_t */
+#define PRIdPTR		"d"	/* intptr_t */
+#define PRIiPTR		"i"	/* intptr_t */
+#define PRIoPTR		"o"	/* intptr_t */
+#define PRIuPTR		"u"	/* uintptr_t */
+#define PRIxPTR		"x"	/* uintptr_t */
+#define PRIXPTR		"X"	/* uintptr_t */
 
 #endif /* INTTYPES__H */
diff --git a/include/lib/libc/aarch64/inttypes_.h b/include/lib/libc/aarch64/inttypes_.h
index 197d627bc..dece3e713 100644
--- a/include/lib/libc/aarch64/inttypes_.h
+++ b/include/lib/libc/aarch64/inttypes_.h
@@ -17,5 +17,12 @@
 #define PRIu64		"lu"	/* uint64_t */
 #define PRIx64		"lx"	/* uint64_t */
 #define PRIX64		"lX"	/* uint64_t */
+#define PRIdPTR		"ld"	/* intptr_t */
+#define PRIiPTR		"li"	/* intptr_t */
+#define PRIoPTR		"lo"	/* intptr_t */
+#define PRIuPTR		"lu"	/* uintptr_t */
+#define PRIxPTR		"lx"	/* uintptr_t */
+#define PRIXPTR		"lX"	/* uintptr_t */
+
 
 #endif /* INTTYPES__H */
diff --git a/include/lib/libc/inttypes.h b/include/lib/libc/inttypes.h
index 0f9e8c612..6ad1c9e10 100644
--- a/include/lib/libc/inttypes.h
+++ b/include/lib/libc/inttypes.h
@@ -17,31 +17,25 @@
 #define PRId8		"d"	/* int8_t */
 #define PRId16		"d"	/* int16_t */
 #define PRId32		"d"	/* int32_t */
-#define PRIdPTR		"d"	/* intptr_t */
 
 #define PRIi8		"i"	/* int8_t */
 #define PRIi16		"i"	/* int16_t */
 #define PRIi32		"i"	/* int32_t */
-#define PRIiPTR		"i"	/* intptr_t */
 
 #define PRIo8		"o"	/* int8_t */
 #define PRIo16		"o"	/* int16_t */
 #define PRIo32		"o"	/* int32_t */
-#define PRIoPTR		"o"	/* intptr_t */
 
 #define PRIu8		"u"	/* uint8_t */
 #define PRIu16		"u"	/* uint16_t */
 #define PRIu32		"u"	/* uint32_t */
-#define PRIuPTR		"u"	/* uintptr_t */
 
 #define PRIx8		"x"	/* uint8_t */
 #define PRIx16		"x"	/* uint16_t */
 #define PRIx32		"x"	/* uint32_t */
-#define PRIxPTR		"x"	/* uintptr_t */
 
 #define PRIX8		"X"	/* uint8_t */
 #define PRIX16		"X"	/* uint16_t */
 #define PRIX32		"X"	/* uint32_t */
-#define PRIXPTR		"X"	/* uintptr_t */
 
 #endif
-- 
2.17.1

